code/modules/mob/abstract/observer/observer.dm

```diff
@@ -355,17 +394,21 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 
 	to_chat(src, "<span class='notice'>Now following \the [following]</span>")
 	move_to_destination(following, following.loc, following.loc)
+	updateghostsight()
 
 /mob/abstract/observer/proc/stop_following()
 	if(following)
 		to_chat(src, "<span class='notice'>No longer following \the [following]</span>")
 		moved_event.unregister(following, src)
 		destroyed_event.unregister(following, src)
 		following = null
+	teleport_if_needed()

```

Comment:
I wouldn't put this here. I'd leave it in the parent calls. We could have fun recursion or double calls otherwise.